scaleway/resource_instance_server.go

```diff
@@ -505,6 +521,32 @@ func resourceScalewayInstanceServerUpdate(d *schema.ResourceData, m interface{})
 		}
 	}
 
+	////
+	// Update reserved IP
+	////
+	if d.HasChange("ip_id") {
+		server, err := instanceAPI.GetServer(&instance.GetServerRequest{
+			Zone:     zone,
+			ServerID: ID,
+		})
+		if err != nil {
+			return err
+		}
+		newIPID := d.Get("ip_id")
+		if server.Server.PublicIP != nil && server.Server.PublicIP.ID != expandID(newIPID) {
+			if newIPID.(string) != "" {
```

Comment:
missing something like:
```
If server.Server.PublicIP.ID != nil {
  Detach IP
}
```

Reply:
Hmm no need, an attach will detach the old IP

Reply:
I agree with @jerome-quere, if user wants to remove the public IP he can't. We need to detach IP if `ip_id` was not empty and it now empty.

Reply:
Yep agree, didn't see it like this

Reply:
fixed